src/openforms/formio/components/custom.py

```diff
@@ -770,6 +771,156 @@ def as_json_schema(component: Component) -> JSONObject:
         return schema
 
 
+class ChildSerializer(serializers.Serializer):
+    bsn = serializers.CharField(
+        label=_("bsn"),
+        max_length=9,
+        help_text=_("The BSN of the child"),
+        validators=[BSNValidator()],
+    )
+    firstNames = serializers.CharField(
+        label=_("firstNames"),
+        help_text=_("The first names of the child"),
+        allow_blank=True,
+    )
+    dateOfBirth = FormioDateField(
+        label=_("date of birth"),
+        help_text=_("The date of birth of the child"),
+    )
+    selected = serializers.BooleanField(
+        label=_("selected"),
+        allow_null=True,
+        help_text=_(
+            "Whether the child is selected by the user or not for further processing"
+        ),
+    )
+
+    def __init__(self, **kwargs):
+        self.component = kwargs.pop("component", None)
+        super().__init__(**kwargs)
+
+
+class ChildListField(serializers.Field):
+    def __init__(self, component, **kwargs):
+        self.component = component
+        super().__init__(**kwargs)
+
+    def to_internal_value(self, data):
+        if not isinstance(data, list):
+            raise serializers.ValidationError("Expected a list of children.")
+
+        serializer = ChildSerializer(
+            data=data,
+            many=True,
+            component=self.component,
+        )
+        serializer.is_valid(raise_exception=True)
+
+        validated = serializer.validated_data
+        self.validate_list(validated)
+        return validated
+
+    def to_representation(self, value):
+        return ChildSerializer(value, many=True, component=self.component).data
+
+    # TODO
+    # Add proper type hints when #2324 is completed (sync the frontend with the backend)
+    def validate_list(self, children):
+        component_key = self.component["key"]
+        submission = self.context["submission"]
+        prefill_data = submission.get_prefilled_data()
+
+        fm_immutable_variable = FormVariable.objects.filter(
+            source=FormVariableSources.user_defined,
+            prefill_plugin=FM_PLUGIN_IDENTIFIER,
+            prefill_options__mutable_data_form_variable=component_key,
+            form=submission.form.id,
```

Comment:
```suggestion
            form=submission.form,
```
it's more pythonic to reason with the model instances than their underlying database ID

Reply:
Apologies, it was there before our last discussion and I forgot about it.